### ISSUE TITLE
Buttons within modals do not always work [2019.0.11-PREVIEW]

### ISSUE THREAD

**Issue by coldwarrl** [Hexworks/zircon#207]
# Buttons within modals seem not always work.

The buttons within modals seem not to work reliable with respect to focus handling and UIEvents.
See example below

```kotlin
@file:Suppress("UNCHECKED_CAST")

package org.hexworks.zircon.internal.integration

import org.hexworks.zircon.api.*
import org.hexworks.zircon.api.builder.component.ModalBuilder
import org.hexworks.zircon.api.extensions.onKeyboardEvent
import org.hexworks.zircon.api.extensions.onMouseEvent
import org.hexworks.zircon.api.uievent.*
import org.hexworks.zircon.internal.component.modal.EmptyModalResult

object ModalTest {

    private val theme = ColorThemes.arc()
    private val tileset = CP437TilesetResources.rexPaint20x20()

    @JvmStatic
    fun main(args: Array<String>) {

        val tileGrid = SwingApplications.startTileGrid(AppConfigs.newConfig()
                .withDefaultTileset(tileset)
                .withSize(Sizes.create(60, 30))
                .build())

        val screen = Screens.createScreenFor(tileGrid)

        val modalPanel = Components.panel()
                .withSize(Sizes.create(20, 20))
                .build()

        val modal = ModalBuilder.newBuilder<EmptyModalResult>()
                .withComponent(modalPanel)
                .withParentSize(tileGrid.size)
                .build()

        val button = Components.button()
                .withText("Close")
                .build().apply {
                    onKeyboardEvent(KeyboardEventType.KEY_PRESSED) { event, _ ->
                        if (event.code == KeyCode.KEY_C) {
                            modal.close(EmptyModalResult)
                            Processed
                        } else Pass
                    }
                }


        button.onMouseEvent(MouseEventType.MOUSE_CLICKED) { event, phase ->
            modal.close(EmptyModalResult)
            Processed
        }
        modalPanel.addComponent(button)

        screen.display()
        screen.applyColorTheme(theme)
        modal.applyColorTheme(theme)

        screen.openModal(modal)
    }

}

```

**Comment by adam-arold** [comment]
Confirmed, I'll fix this tomorrow.

**Comment by adam-arold** [comment]
Fixed in [2019.0.12-PREVIEW](https://jitpack.io/#org.hexworks/zircon/2019.0.12-PREVIEW)
